packages/flutter/test/widgets/actions_test.dart

```diff
@@ -951,6 +966,41 @@ void main() {
         expect(buttonNode2.hasFocus, isFalse);
       },
     );
+
+    testWidgets('FocusableActionDetector can exclude Focus semantics', (WidgetTester tester) async {
+      await tester.pumpWidget(
+        const MaterialApp(
+          home: FocusableActionDetector(
+            child: SizedBox(
+              width: 50,
+              height: 50,
+            ),
+          ),
+        )
+      );
+
+      expect(
+        tester.getSemantics(find.byType(Focus).last),
+        matchesSemantics(isFocusable: true),
+      );
+
+      await tester.pumpWidget(
+        const MaterialApp(
+          home: FocusableActionDetector(
+            includeFocusSemantics: false,
+            child: SizedBox(
+              width: 50,
+              height: 50,
+            ),
+          ),
+        )
+      );
+
+      expect(
+        tester.getSemantics(find.byType(Focus).last),
+        matchesSemantics(scopesRoute: true),
+      );
```

Comment:
If this is doing what I think it is, this expectation is pretty misleading. From what I can gather, this expectation is saying "There is no SemanticsNode directly related to `Focus`, instead the `Focus` semantics are handled by the outer `MaterialApp`'s SemanticsNode.".

Could we make these two expectations more clear by using the `children` parameter of `matchesSemantics`, or at a minimum add a `reason` explaining what's going on for the second expectation?

Reply:
Thank you so much for the feedback. I agree the test wasn't obvious. 

I have updated the tests and added comments with a better example for `children`. 
Let me know if this is clear enough.

Reply:
Looks great, thanks!